src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -203,40 +201,47 @@ public void unselectDeck() {
 
     @Override
     public Optional<Deck> getSelectedDeck() {
-        return selectedDeck;
+        return Optional.ofNullable(selectedDeck);
     }
 
     @Override
     public String getSelectedDeckName() {
-        return selectedDeck.get().getDeckName();
+        return Optional.ofNullable(selectedDeck)
+                .map(Deck::getDeckName)
+                .orElse("None");
```

Comment:
Should we return "None" here? I think can consider doing away the `orElse("None")`.
Ie we just do `Optional.ofNullable(selectedDeck).map(Deck::getDeckName)` since this will return an empty Optional object which we can apply `isPresent` if we require to check if there is a value.

Reply:
ya agreed

Reply:
I agree, though initially I thought of handling null here as a sanity check only. The only usage of `getSelectedDeckName` right now is in SelectDeckCommand, and I am not sure if handing null check there violates any SLAP or abstraction principles. Will modify if you guys agree there is no violation!

![image](https://user-images.githubusercontent.com/80334013/224910718-84847a35-8543-4ab2-8dfa-199add35787f.png)